Source: Ingmar Vanhassel <ingmar@exherbo.org>
Upstream: No
Reason: If we explicitly enable a Package, we want to be told that it wasn't found.
    This patch makes this an error, if WITH_STRICT_DEPENDENCIES is enabled.

From 9a0decb405dce422286af0fde89f67307d2db086 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Fri, 24 Apr 2009 00:27:32 +0200
Subject: [PATCH] Add WITH_STRICT_DEPENDENCIES

---
 cmake/modules/MacroOptionalFindPackage.cmake |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/cmake/modules/MacroOptionalFindPackage.cmake b/cmake/modules/MacroOptionalFindPackage.cmake
index 816cdb7..11571ac 100644
--- a/cmake/modules/MacroOptionalFindPackage.cmake
+++ b/cmake/modules/MacroOptionalFindPackage.cmake
@@ -16,7 +16,12 @@
 MACRO (MACRO_OPTIONAL_FIND_PACKAGE _name )
    OPTION(WITH_${_name} "Search for ${_name} package" ON)
    if (WITH_${_name})
-      FIND_PACKAGE(${_name} ${ARGN})
+      OPTION(WITH_STRICT_DEPENDENCIES "Handle optional dependencies more strictly" ON)
+      if (WITH_STRICT_DEPENDENCIES)
+          FIND_PACKAGE(${_name} ${ARGN} REQUIRED)
+      else (WITH_STRICT_DEPENDENCIES)
+          FIND_PACKAGE(${_name} ${ARGN})
+      endif (WITH_STRICT_DEPENDENCIES)
    else (WITH_${_name})
       set(${_name}_FOUND)
       set(${_name}_INCLUDE_DIR)
-- 
1.6.3

